### PR TITLE
Support Numba-like `jit.laneid()` and `jit.warpsize` syntax in CuPy JIT

### PR DESCRIPTION
**Pull request by leofang**
Follow-up of #5334 and #5461. ~~To be rebased after #5461 is merged.~~ 

Note that unlike in Numba, `laneid()` has to be a callable. The reason is that IIUC only the `BuiltinFunc` subclasses have access to `env`, which would allow us to insert a preamble as needed. I'd be happy to learn if there's any alternative approach...

### CODE REVIEW

**Comment by leofang** [comment]
Merged with master!

File: cupyx/jit/_interface.py [hunk 1]
```diff
@@ -180,3 +180,16 @@ def __init__(self, name):
 blockDim = _Dim3('blockDim')
 blockIdx = _Dim3('blockIdx')
 gridDim = _Dim3('gridDim')
+
+warpsize = _internal_types.Data(
+    '64' if runtime.is_hip else '32', _cuda_types.uint32)
```
Comment (asi1024) [inline cupyx/jit/_interface.py:185]: `warpsize` can be defined as a constant value.
```suggestion
warpsize = numpy.int32(64'if runtime.is_hip else 32)
```

Comment (leofang) [reply cupyx/jit/_interface.py:185]: @asi1024 Interesting, thanks. But, I found that in this case we cannot overwrite `warpsize`'s `__doc__` attribute:
```python
>>> a = np.uint32(10)
>>> a.__doc__ = "hahaha"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'numpy.uint32' object attribute '__doc__' is read-only
```
Moreover, this doesn't work with RTD either (tried in fbf602b):
```python
warpsize = ...
"""my docstring"""
```
Am I missing something?

Comment (asi1024) [reply cupyx/jit/_interface.py:185]: I got the point. Thanks!

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
LGTM. Thanks!

**Comment by leofang** [comment]
Thank you, @asi1024!

**Pull request merged**